linode/kernel/framework_datasource.go

```diff
@@ -0,0 +1,100 @@
+package kernel
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/linode/linodego"
+	"github.com/linode/terraform-provider-linode/linode/helper"
+)
+
+func NewDataSource() datasource.DataSource {
+	return &DataSource{}
+}
+
+type DataSource struct {
+	client *linodego.Client
+}
+
+func (data *DataSourceModel) parseKernel(kernel *linodego.LinodeKernel) {
+	data.ID = types.StringValue(kernel.ID)
```

Comment:
nit: I think we probably don't have to set the `ID` value here, because it's already grabbed and set from `req.Config`.

Reply:
Fixed. Thanks for the tip!